vms/avm/states/state.go

```diff
@@ -563,6 +579,8 @@ func (s *state) writeTxs() error {
 		txID := txID
 		txBytes := tx.Bytes()
 
+		s.updateTxChecksum(txID)
```

Comment:
We write all txs we are accepting to disk. Even transactions that were _already_ written when they were processing. While the following `db.Put` will be a noop in this case - the DB will also end up deleting the status entry... Which marks this tx as accepted. Regardless - it is correct to update the checksum here.